assets/javascripts/openqa.js

```diff
@@ -290,28 +290,45 @@ function renderSearchResults(query, url) {
     }
     spinner.style.display = 'none';
     var heading = document.getElementById('results-heading');
-    heading.appendChild(document.createTextNode(': ' + json.data.length + ' matches found'));
+    heading.appendChild(document.createTextNode(': ' + json.data.total_count + ' matches found'));
     var results = document.createElement('div');
     results.id = 'results';
     results.className = 'list-group';
-    json.data.forEach(function (value, index) {
-      var item = document.createElement('div');
-      item.className = 'list-group-item';
-      var header = document.createElement('div');
-      header.className = 'd-flex w-100 justify-content-between';
-      var title = document.createElement('h5');
-      title.className = 'occurrence mb-1';
-      title.appendChild(document.createTextNode(value.occurrence));
-      header.appendChild(title);
-      item.appendChild(header);
-      if (value.contents) {
-        var contents = document.createElement('pre');
-        contents.className = 'contents mb-1';
-        contents.appendChild(document.createTextNode(value.contents));
-        item.appendChild(contents);
+    const types = ['code', 'modules', 'templates'];
+
+    for (var i = 0; i < types.length; i++) {
+      var searchtype = types[i];
+      var searchresults = json.data.results[searchtype];
+      if (searchresults.length > 0) {
+        var header = document.createElement('div');
+        header.id = searchtype;
+        var bold = document.createElement('strong');
+        var textnode = document.createTextNode(searchtype + ': ' + searchresults.length);
+        bold.appendChild(textnode);
+
+        header.className = 'list-group-item';
+        header.appendChild(bold);
+        results.append(header);
       }
-      results.append(item);
-    });
+      searchresults.forEach(function (value, index) {
+        var item = document.createElement('div');
```

Comment:
Can't many of these `var` uses be replaced with `const`?

Reply:
Ok, I replaced them, also in the existing loop